File: Programing_Basics_with_JS/exam28And29May2022/03.SantasHoliday.js
function santasHoliday(input) {
    let days = Number(input[0]);
    let kindOfRoom = input[1];
    let grade = input[2];
    
    let nightCost = 0;
    let roomPrice = 18
    let apartmentPrice = 25;
    let presidentApartmentPrice = 35;
    let night = days - 1;

    switch(kindOfRoom) {
        case 'room for one person': nightCost = night * roomPrice; break;
        case 'apartment': 
        nightCost = night * apartmentPrice; 
            if(days < 10) {
                nightCost = nightCost * 0.70;
            } else if (days <= 15) {
                nightCost = nightCost * 0.65;
            } else {
                nightCost = nightCost * 0.50;
            }
        break;

        case 'president apartment': 
        nightCost = night * presidentApartmentPrice; 
        if(days < 10) {
            nightCost = nightCost * 0.90;
        } else if (days <= 15) {
            nightCost = nightCost * 0.85;
        } else {
            nightCost = nightCost * 0.80;
        }
        break; 
    }

    let totalCost = 0;

    if(grade === 'positive') {
        totalCost = nightCost * 1.25;
    } else if (grade === 'negative') {
        totalCost = nightCost * 0.9;
    }

    console.log(totalCost.toFixed(2));
}

santasHoliday(["14",
"apartment",
"positive"])

santasHoliday(["30",
"president apartment",
"negative"])

santasHoliday(["12",
"room for one person",
"positive"])

santasHoliday(["2",
"apartment",
"positive"])

File: Programming_Basics_with_Python/programming_basics_online_pre_exam_22_and_23_october_2022/03._santas_holiday.py
days = int(input())
type_of_room = input()
score = input()

nights = days - 1
room_price = 0

if type_of_room == 'room for one person':
    room_price = nights * 18.00
elif type_of_room == 'apartment':
    room_price = nights * 25.00
    if days < 10:
        room_price = room_price * 0.70
    elif 10 <= days <= 15:
        room_price = room_price * 0.65
    elif days > 15:
        room_price = room_price * 0.50
elif type_of_room == 'president apartment':
    room_price = nights * 35.00
    if days < 10:
        room_price = room_price * 0.90
    elif 10 <= days <= 15:
        room_price = room_price * 0.85
    elif days > 15:
        room_price = room_price * 0.80

if score == 'positive':
    room_price = room_price * 1.25
elif score == 'negative':
    room_price = room_price * 0.9

print(f'{room_price:.2f}')